src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -953,7 +953,7 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 			// Respect chords if the allowChords setting is set and it's not Escape. Escape is
 			// handled specially for Zen Mode's Escape, Escape chord, plus it's important in
 			// terminals generally
-			const isValidChord = resolveResult?.kind === ResultKind.MoreChordsNeeded && this._configHelper.config.allowChords && event.key !== 'Escape';
+			const isValidChord = resolveResult.kind === ResultKind.MoreChordsNeeded && this._configHelper.config.allowChords && event.key !== 'Escape';
```

Comment:
Hi @Tyriar 

I refactored the keybindings resolver and, consequently, `AbstractKeybindingService#softDispatch` a bit in last iteration and propagated the changes to various parts of vscode that use the softDispatch. I introduced a regression in one such place and now trying to make sure my changes in other places make sense. 

Would you mind confirming that what this line does matches the desired result? This particular PR doesn't do much, there's a more meaningful diff here - https://github.com/microsoft/vscode/pull/178699/files#